### PR TITLE
rename l_function method name

### LINKED ISSUE #210: E743 ambiguous function definition
```py
    def l(self, s):
        """
        Evaluates the Dirichlet L-function of this character at the given
        complex number s.

            >>> from flint import showgood
            >>> chi = dirichlet_char(1, 1)
            >>> showgood(lambda: chi.l(2), dps=25)
            1.644934066848226436472415
            >>> chi = dirichlet_char(7, 3)
            >>> showgood(lambda: chi.l(2+3j), dps=25)
            1.273313649440490751755284 - 0.07432329442559421607102118j

        """
        s = any_as_acb(s)
        cdef acb v
        v = acb.__new__(acb)
        acb_dirichlet_l((<acb>v).val, (<acb>s).val, self.G.val, self.val, getprec())
        return v
```

We currently have this function, which triggers the listing error `E741 ambiguous variable name 'l'`. I think we could probably call this `dirichlet_l` to match Flint, or `l_function`? I'm not sure what other people think so I've made this issue to discuss.

### PR DESCRIPTION
**Pull request by GiacomoPope**
Fixes #210

### CODE REVIEW

File: src/flint/types/dirichlet.pyx [hunk 1]
```diff
@@ -189,6 +189,9 @@ cdef class dirichlet_char(object):
         acb_dirichlet_l((<acb>v).val, (<acb>s).val, self.G.val, self.val, getprec())
         return v
 
+    # For backwards compatibility we allow self.l(s) see Issue #210
+    l = l_function # no-cython-lint
```
Comment (oscarbenjamin) [inline src/flint/types/dirichlet.pyx:193]: There isn't any point in doing this unless the docstring is updated to suggest calling the new function name.

I would do it like:
```
def l_function(self, s):
    """
    >>> l_function()
    123
    """
    ...


def l(self, s):
    """
    Alias for :meth:`l_function`
    """
    return self.l_function(s)
```

**Comment by oscarbenjamin** [comment]
I'm not sure it is worth changing the name here. I don't know much about this function and I don't know what it would normally be called. If the linter wasn't complaining then it would not have been suggested to change this so I would probably just leave it I think.

Putting "function" in the name seems sort of redundant when most methods are functions.

File: src/flint/types/dirichlet.pyx (hunk 1 above)
Comment (GiacomoPope) [reply src/flint/types/dirichlet.pyx]: oh sorry, i just missed this

**Comment by GiacomoPope** [comment]
I think `l_function` makes sense, as it's a mathematical object known as a l-function (https://en.wikipedia.org/wiki/Dirichlet_L-function), obviously all the code is built with methods/functions but just calling it `l` doesnt seem very helpful to me. Happy to just close this though if you disagree

**Comment by oscarbenjamin** [comment]
Okay, let's get it in.

**Pull request merged**